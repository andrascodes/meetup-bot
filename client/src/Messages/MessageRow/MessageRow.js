import React from 'react'

import { Message } from './Message'

export const MessageRow = props => {
  let IncomingMessage = ''
  let OutgoingMessage = ''
  if(props.direction === 'incoming') {
    IncomingMessage = (
      <Message {...props}/>
    )
  }
  else {
    OutgoingMessage = (
      <Message {...props}/>
    )
  }

  return (
    <div className="row">
      <div className="col-lg-1 col-md-1"></div>
      <div className="col-lg-5 col-md-5">
        {IncomingMessage}
      </div>
      <div className="col-lg-5 col-md-5">
        {OutgoingMessage}
      </div>
      <div className="col-lg-1 col-md-1"></div>
    </div>
  )
}