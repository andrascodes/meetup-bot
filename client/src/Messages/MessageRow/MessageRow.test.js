import React from 'react'
import renderer from 'react-test-renderer'

import { MessageRow } from './MessageRow'

describe('MessageRow', () => {

  test('renders Incoming Message correctly', () => {
    const tree = renderer.create(
      <MessageRow
        timestamp={'2017-04-10T15:16:55.834Z'}
        type={'text'}
        direction={'incoming'} 
        message={{
          message: {
            text: 'Incoming text message'
          }
        }}
        text={'Text: Incoming text message'}
        response={{
          error: {
            code: 200,
          }
        }}
        error={'OAuthException'}
        sentiment={'neutral'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('renders Outgoing Message correctly', () => {
    const tree = renderer.create(
      <MessageRow
        timestamp={'2017-04-10T15:16:55.834Z'}
        type={'text'}
        direction={'outgoing'} 
        message={{
          message: {
            text: 'Incoming text message'
          }
        }}
        text={'Text: Incoming text message'}
        response={{
          error: {
            code: 200,
          }
        }}
        error={'OAuthException'}
        sentiment={'neutral'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})