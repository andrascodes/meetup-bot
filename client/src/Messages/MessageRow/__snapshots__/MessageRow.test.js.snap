exports[`MessageRow renders Incoming Message correctly 1`] = `
<div
  className="row">
  <div
    className="col-lg-1 col-md-1" />
  <div
    className="col-lg-5 col-md-5">
    <div>
      <div
        className="TimeIndicator text-center">
        Arrived at 
        17:16:55
      </div>
      <div>
        <div
          className="Incoming Bubble ErrorBubble"
          data-tip="React tooltip">
          <div
            className="TextMessage">
            Incoming text message
          </div>
        </div>
        <div
          className="ErrorTooltip">
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip">
            
          </div>
        </div>
      </div>
      <div
        className="row Indicators">
        <div
          className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-left">
          <div
            className="Sentiment">
            Sentiment: 
            <span
              className="Neutral">
              neutral
            </span>
            <i
              className="fa fa-check Neutral Icon" />
          </div>
        </div>
        <div
          className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-right">
          <div
            className="ErrorIndicator">
            Error: 
            <span
              className="ErrorStyle">
              OAuthException #200
            </span>
            <i
              className="fa fa-times ErrorStyle Icon" />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-lg-5 col-md-5">
    
  </div>
  <div
    className="col-lg-1 col-md-1" />
</div>
`;

exports[`MessageRow renders Outgoing Message correctly 1`] = `
<div
  className="row">
  <div
    className="col-lg-1 col-md-1" />
  <div
    className="col-lg-5 col-md-5">
    
  </div>
  <div
    className="col-lg-5 col-md-5">
    <div>
      <div
        className="TimeIndicator text-center">
        Sent at 
        17:16:55
      </div>
      <div>
        <div
          className="Outgoing Bubble ErrorBubble"
          data-tip="React tooltip">
          <div
            className="TextMessage">
            Incoming text message
          </div>
        </div>
        <div
          className="ErrorTooltip">
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip">
            
          </div>
        </div>
      </div>
      <div
        className="row Indicators">
        <div
          className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-left">
          <div
            className="Sentiment">
            Sentiment: 
            <span
              className="Neutral">
              neutral
            </span>
            <i
              className="fa fa-check Neutral Icon" />
          </div>
        </div>
        <div
          className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-right">
          <div
            className="ErrorIndicator">
            Error: 
            <span
              className="ErrorStyle">
              OAuthException #200
            </span>
            <i
              className="fa fa-times ErrorStyle Icon" />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-lg-1 col-md-1" />
</div>
`;
